### PR TITLE
Iterate expired components from the da_checker

### PR DESCRIPTION
**Pull request by dapplion**
## Issue Addressed

Closes https://github.com/sigp/lighthouse/issues/5892

## Proposed Changes

On the pruning routine, iterate expired entries both when threshold is reached and when it's not

### CODE REVIEW

**Comment by michaelsproul** [comment]
Thoughts on including this in 5.2 if we are about to re-test over the weekend?

File: beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs [hunk 1]
```diff
@@ -776,6 +776,21 @@ impl<T: BeaconChainTypes> OverflowLRUCache<T> {
...
+        for (key, value) in write_lock.in_memory.iter() {
+            if let Some(epoch) = value.epoch() {
+                if epoch < cutoff_epoch {
+                    keys_to_remove.push(*key);
+                }
+            }
+        }
+        // Now remove keys
+        for key in keys_to_remove {
+            write_lock.in_memory.pop(&key);
```
Comment (michaelsproul) [inline beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs]: this just stops the cache from building up but won't clear old junk that's already on disk, will it?

**Comment by michaelsproul** [comment]
On 2nd thoughts I don't think this is worth including in 5.2. Let's get 5.2 done and then we can consider a follow-up release

File: beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs (hunk 1 above)
Comment (dapplion) [reply beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs]: There's a second maintenance routine that prunes keys in disk that are not in the memory map

**Comment by michaelsproul** [comment]
Do we still need this now that the overflow cache is gone?

- https://github.com/sigp/lighthouse/pull/5891

**Comment by dapplion** [comment]
> Do we still need this now that the overflow cache is gone?

Yes, the in-memory map has a leak that is independent to the disk portion that just got dropped.

**Review by realbigsean — APPROVED** [review]
LGTM

**Review by ethDreamer — APPROVED** [review]
looks good :)

**Comment by realbigsean** [comment]
@mergify queue

**Pull request merged**